### ISSUE TITLE
Multiple users can edit specific Article

### ISSUE THREAD

**Issue by BojanKogoj** [Zizaco/entrust#444]
Is it possible to give several users permission to edit a specific article? Only that one, not all.

**Comment by lcdss** [comment]
Yes, but somehow you need associate your article model with permission model. You can save many permissions ("article.{id}.edit", "article.id.delete", "article.id.show") when a new article has been created. So you can always verify what permissions an user has with a specified article.

**Comment by BojanKogoj** [comment]
And what if several user can edit multiple articles? Could I make a group? How would I be able to retrieve them? Unless I looped through them and extracting id from string.

**Comment by lcdss** [comment]
That {id} is the article id, not user id.

**Comment by BojanKogoj** [comment]
This could work. But how would I make groups? Several users have access to edit several articles, all the same. Another group can edit some other articles.

**Comment by BojanKogoj** [comment]
Ok, managed to figure it out, I was using https://github.com/romanbican/roles before which works a bit differently, so it confused me.

**Comment by BojanKogoj** [comment]
@lcdss unfortunately your solution doesn't work.
```
'middleware' => [permission:article.{id}.edit'],
```
doesn't work, it throws 403. It however works if I replace {id} with an actual id it works. Are middleware arguments not getting replaced by value?

**Comment by lcdss** [comment]
I never tried it, but yes, it works if you do right (i guess).

The user 'Taylor' creates the article 'My dog is awesome'. (id = 666)
Now we create one permission or more, depending on your needs.
$role = Role::create(['article.666', 'Permission to modify (update or delete) the article id 666']);
Auth::user()->attachPermission($role);

Now you can attach the permission with any user and check if an user has this permission.

**Comment by BojanKogoj** [comment]
The problem with this is that I can't change {id} dynamically. I can't have a static 'article.666' in route. 
I created a thread and partial solution [on laravel forums](http://laravel.io/forum/12-21-2015-middleware-with-parameters), but we could add this parameter parser to Entrust, it would make my life easy!

**Comment by lcdss** [comment]
I think you are trying to do this in the wrong place. You could use Form Request, Controller, Middleware or a Policy class, but not in route, because it's not suitable for static restrictions.

**Comment by BojanKogoj** [comment]
Yeah, I've done it in middleware. But I could really use the method visible on the link above in EntrustPermission.php, so all permissions would be fixed with actual id there
